src/components/Task/Modal/TaskModal.tsx

```diff
@@ -45,69 +52,63 @@ const TaskModal = ({
 
   const task = data?.task;
 
-  const renderTitle = (): string => {
+  const renderTitle = useMemo((): string => {
```

Comment:
If this `useMemo` a performance optimization? Are you sure it makes much of a difference? I'd be surprised if executing a `switch` is more expensive than the overhead of `useMemo`.

Reply:
Adding the useMemo on the `renderTitle` was "while I'm here" kind of work. It probably won't make much of a difference.

Reply:
It's slower on the first render and faster on subsequent renders as expected, but also we're talking about no more than 10s of microseconds.

The first pair is the first render, subsequent pairs are future renders.

<img width="247" alt="Screenshot 2024-04-25 at 12 33 02 PM" src="https://github.com/CruGlobal/mpdx-react/assets/3740187/83576643-be4a-4e3a-8107-3b5a1b1814c5">

Reply:
How did you get that view of the Memoized vs Not Memoized? That's a great feature

Reply:
I just wrapped it in `console.time/console.timeEnd`.

```ts
  console.time('Memoized');
  const renderTitle = useMemo((): string => {
    switch (view) {
      case TaskModalEnum.Complete:
        return t('Complete Task');
      case TaskModalEnum.Comments:
        return t('Task Comments');
      case TaskModalEnum.Log:
        return t('Log Task');
      case TaskModalEnum.Edit:
        return t('Edit Task');
      default:
        return t('Add Task');
    }
  }, [view]);
  console.timeEnd('Memoized');

  console.time('Not memoized');
  let title;
  switch (view) {
    case TaskModalEnum.Complete:
      title = t('Complete Task');
      break;
    case TaskModalEnum.Comments:
      title = t('Task Comments');
      break;
    case TaskModalEnum.Log:
      title = t('Log Task');
      break;
    case TaskModalEnum.Edit:
      title = t('Edit Task');
      break;
    default:
      title = t('Add Task');
      break;
  }
  console.timeEnd('Not memoized');
```

Reply:
I'm going to stick with the memo. I did some more testing and found the function was called less often. I tested this by putting a console log in the function and counting the times is was logged out.
With the memorized function, the function was only logged twice (both when the modal opens), and with the un-memorized function, it was logged 6 times (4 when the modal opens and 2 when closes)

Reply:
Definitely! That's what I would expect to see. I assume it's called twice instead of once because of React strict mode.